### PR TITLE
3475 0

### PR DESCRIPTION
**Pull request by ButterBrot777**
## Номер
3475

## Номер задания
0

## Ссылка на видео с демо
https://www.youtube.com/watch?v=ecIWPzGEbFc

## Комментарии
Пришлось создать новый ПР, т.к. старый перестал открываться (долго загружался), хотя все работало.

### CODE REVIEW

**Comment by longbeardmen** [comment]
change PR name please. no one cares if it's new or old one

**Comment by longbeardmen** [comment]
Commits history looks weird and tells nothing.
In this particular case I would squash them in one

File: 3475/0/about_sandwich_code.rb [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  end
+
+  def test_finding_lines_two
+    assert_equal "test\n", find_line_two('example_file.txt')
+  end
+
+  # ------------------------------------------------------------------
+  # :reek:UtilityFunction
+  # :reek:UncommunicativeMethodName
+  # rubocop:disable all
```
Comment (Moleved) [inline 3475/0/about_sandwich_code.rb]: Do not disable all

File: 3737/0/about_java_interop.rb [hunk 2]
```diff
@@ -1,8 +1,9 @@
+# rubocop:disable all
```
Comment (Moleved) [inline 3737/0/about_java_interop.rb]: Why this file is here?

File: 3737/0/about_java_interop.rb [hunk 3]
```diff
@@ -13,21 +14,20 @@
 # * Calling custom java class
 # * Calling Ruby from java???
 
-#:nodoc:
 class AboutJavaInterop < Neo::Koan
   def test_using_a_java_library_class
     java_array = java.util.ArrayList.new
-    assert_equal Java::JavaUtil::ArrayList, java_array.class
+    assert_equal __, java_array.class
```
Comment (longbeardmen) [inline 3737/0/about_java_interop.rb]: there should be no `__` in asserts

File: 3737/0/about_java_interop.rb [hunk 4]
```diff
@@ -13,21 +14,20 @@
...
   end
 
   def test_java_class_can_be_referenced_using_both_ruby_and_java_like_syntax
-    assert_equal true, Java::JavaUtil::ArrayList == java.util.ArrayList
+    assert_equal __, Java::JavaUtil::ArrayList == java.util.ArrayList
```
Comment (longbeardmen) [inline 3737/0/about_java_interop.rb]: read about `asset_equal`
if you want to check if objects are equal you should pass them as params
```
 => a == b
true
```

`assert_equal a, b`

File: 3475/0/about_class_methods.rb [hunk 5]
```diff
@@ -0,0 +1,172 @@
+require File.expand_path(File.dirname(__FILE__) + '/neo')
+
+# This method smells of :reek:TooManyMethods
+# This method smells of :reek:UncommunicativeMethodName
+class AboutClassMethods < Neo::Koan
+  class Dog
```
Comment (longbeardmen) [inline 3475/0/about_class_methods.rb]: empty classes should be defined in one line as `class Foo; end`

File: 3475/0/about_control_statements.rb [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+
+      result << i
+    end
+    assert_equal [1, 3, 5, 7, 9], result
+  end
+
+  def test_for_statement
+    array = %w[fish and chips]
+    result = []
+    array.each do |item|
```
Comment (longbeardmen) [inline 3475/0/about_control_statements.rb]: where's for statement in this test?!!

**Comment by longbeardmen** [comment]
откатить все изменения после `rubocop -a`
отключать копы, если это действительно нужно

**Comment by Moleved** [comment]
Remove .swp file from PR

**Comment by Moleved** [comment]
Fix comments of previous reviewer

File: 3475/0/versioning_spec.rb [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+def version_ints(version)
+  version.split('.').map(&:to_i)
+end
+
+# This method smells of :reek:NilCheck
+# This method smells of :reek:UncommunicativeVariableName
+def at_least_ruby_version(version)
+  vints = version_ints(version)
+  ruby_vints = version_ints(RUBY_VERSION)
+  vints.zip(ruby_vints).all? { |v, rv| v.nil? || rv.nil? || v >= rv }
```
Comment (longbeardmen) [inline 3475/0/versioning_spec.rb]: `vints` variable name doesn't make any sense without context as well as `v` and `rv`.
Variable name should utilize fill word

File: 3475/0/triangle.rb [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  elsif site1 == site2 || site1 == site3 || site2 == site3
+    :isosceles
+  else
+    :scalene
+  end
+end
+
+# :reek:UncommunicativeParameterName
+# :reek:FeatureEnvy
+def invalid_traingle?(site1, site2, site3)
```
Comment (longbeardmen) [inline 3475/0/triangle.rb]: `side`

File: 3475/0/about_array_assignment.rb [hunk 9]
```diff
@@ -0,0 +1,62 @@
+require File.expand_path(File.dirname(__FILE__) + '/neo')
+
+class AboutArrayAssignment < Neo::Koan
+  def test_non_parallel_assignment
+    names = %w[John Smith]
+    assert_equal %w[John Smith], names
+  end
+
+  # rubocop:disable Style/WordArray
```
Comment (longbeardmen) [inline 3475/0/about_array_assignment.rb]: this should be fixed

**Pull request merged**